commerce_coordinator/apps/iap/views.py

```diff
@@ -0,0 +1,123 @@
+"""
+Views for the InAppPurchase app
+"""
+
+import datetime
+import logging
+import uuid
+
+from commercetools import CommercetoolsError
+from commercetools.platform.models import Money
+from iso4217 import Currency
+from rest_framework import status
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+# isort: off
+from commerce_coordinator.apps.commercetools.catalog_info.constants import TwoUKeys
+from commerce_coordinator.apps.commercetools.clients import CommercetoolsAPIClient
+from commerce_coordinator.apps.iap.utils import (
+    get_ct_customer,
+    get_email_domain,
+    get_standalone_price_for_sku,
+)
+from commerce_coordinator.apps.iap.serializers import (
+    MobileOrderRequestData,
+    MobileOrderRequestSerializer,
+)
+
+# isort: on
+
+logger = logging.getLogger(__name__)
+
+
+class MobileCreateOrderView(APIView):
+    """
+    API view for preparing a cart in CT and then converting it to an order
+    for mobile In-App purchase
+    """
+
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request: Request) -> Response:
+        """
+        Handles POST request for preparing a cart in CT and then converting it
+        to an order for mobile In-App purchase
+        """
+        try:
+            serializer = MobileOrderRequestSerializer(data=request.data)
+            serializer.is_valid(raise_exception=True)
+            data = MobileOrderRequestData(**serializer.validated_data)  # type: ignore
+
+            fraction_digits = Currency(data.currency_code).exponent or 0
+            external_price = Money(
+                cent_amount=int(data.price.scaleb(fraction_digits)),
+                currency_code=data.currency_code,
+            )
+            standalone_price = get_standalone_price_for_sku(
+                sku=data.course_run_key,
+            )
+
+            client = CommercetoolsAPIClient(enable_retries=True)
+            customer = get_ct_customer(client, request.user)
```

Comment:
@NawfalAhmed @attiyaIshaque Please make sure that if an exception is raised at this point, we do not need to delete the mobile cart since it has not been created yet.